packages/agents-realtime/src/openaiRealtimeBase.ts

```diff
@@ -406,6 +408,48 @@ export abstract class OpenAIRealtimeBase
     return sessionData;
   }
 
+  private static buildTurnDetectionConfig(
+    c: RealtimeTurnDetectionConfig | undefined,
+  ): RealtimeTurnDetectionConfigAsIs | undefined {
+    if (typeof c === 'undefined') {
+      return undefined;
+    }
+    const {
+      type,
+      createResponse,
+      create_response,
+      eagerness,
+      interruptResponse,
+      interrupt_response,
+      prefixPaddingMs,
+      prefix_padding_ms,
+      silenceDurationMs,
+      silence_duration_ms,
+      threshold,
+      ...rest
+    } = c;
+
+    const config: RealtimeTurnDetectionConfigAsIs & Record<string, any> = {
+      type,
+      create_response: createResponse ? createResponse : create_response,
+      eagerness,
+      interrupt_response: interruptResponse
+        ? interruptResponse
+        : interrupt_response,
+      prefix_padding_ms: prefixPaddingMs ? prefixPaddingMs : prefix_padding_ms,
+      silence_duration_ms: silenceDurationMs
+        ? silenceDurationMs
+        : silence_duration_ms,
+      threshold,
+      ...rest,
+    };
+    // Remove undefined values from the config
```

Comment:
When I verified the behavior, having undefined values could affect the connection establishment, so I added this logic. but if my observation is wrong or is missing something, please feel free to adjust this part.